### PR TITLE
Pull Request: Implement CIFAR-10 Data Loading with Augmentation and Normalization

### LINKED ISSUE #1: Implement Dataset loading
In `data/dataloader.py`, importing of the CIFAR-10 Dataset should be implemented.

- Use the datasets from the `torchvision` module
- Load the train_dataset and test_dataset separately (by setting the `train=True/False` parameter)
- Use `download=True` as well as the `data` directory as `root`
- Use the data transforms returned by `get_train_transform()` and `get_test_transform()`, which will be implemented in a separate Issue

Put this into a `load_data()` function that returns the dataloaders, where `batch_size` is given as an argument. Shuffle training, but not test data.

### LINKED ISSUE #2: Implement Dataset Normalization
In `data/dataloader.py`, both training and testing data should be normalized for training the GAN with Tanh output. (MEAN and STD set to (0.5,0.5,0.5) each)

Return the final transforms in the `get_train_transform()` and `get_test_transform()` functions respectively.

### LINKED ISSUE #3: Implement Data Augmentation
In `data/dataloader.py`, increase the diversity of our training data by applying several transformations:

- RandomHorizontalFlip
- RandomRotation
- ColorJitter (? might not be suitable for this type of images)
- RandomResizedCrop

inside of the `get_train_transform()` and `get_test_transform()` functions.

### PR DESCRIPTION
**Pull request by 0xBA11E5**
### This PR implements the core data loading and parts of the processing architecture for the CIFAR-10 dataset

- DataLoader Architecture: Established the primary data loading method and logic, allowing for modular configuration.
- Data Augmentation: Integrated RandomFlip and RandomRotation for the training set to increase sample diversity.
- Dataset Normalization: Included a transformation to scale and normalize pixel values from [0,1] to the [−1,1] (using the specified normalization parameters).
- Configuration & Testing: Added a minimal working example to verify the pipeline's functionality.
- Environment: Updated requirements.txt to include necessary dependencies for the new data loading logic.

This PR completes the following roadmap items:
Resolves #1: Implement Dataset loading
Resolves #2: Implement Dataset Normalization
Resolves #3: Implement Data Augmentation

### CODE REVIEW

File: data/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    #
+    # train_l, test_l = get_dataloaders(config)
+    #
+    # # Check one batch
+    # images, labels = next(iter(train_l))
+    # print(f"Batch shape: {images.shape}")  # Should be [64, 3, 32, 32]
+    # print(f"Labels shape: {labels.shape}")
+    # print(
+    #     f"Min val: {images.min():.2f}, Max val: {images.max():.2f}"
+    )  # Should be approx -1 and 1
```
Comment (Supergecki) [inline data/dataloader.py]: There is a `#` missing at the start of this line

**Review by Supergecki — APPROVED** [review]
Works now, will merge.
